python/pyspark/pandas/series.py

```diff
@@ -4475,6 +4477,146 @@ def replace(
+        duck       30.0
+        eagle     200.0
+        falcon    345.0
+        dtype: float64
+        >>> reset_option("compute.ops_on_diff_frames")
+        """
+        if not isinstance(other, Series) and not np.isscalar(other):
+            raise TypeError("unsupported type: %s" % type(other))
+
+        tmp_other_col = verify_temp_column_name(self._internal.spark_frame, "__tmp_other_col__")
+        if np.isscalar(other):
+            combined = DataFrame(
+                InternalFrame(
+                    spark_frame=self._internal.spark_frame.select(
+                        *self._internal.spark_columns
+                    ).withColumn(tmp_other_col, SF.lit(other)),
+                    index_spark_columns=self._internal.index_spark_columns,
+                )
+            )
+        elif same_anchor(self, other):
+            combined = self._psdf[self._column_label, other._column_label]
+        elif fill_value is None:
+            combined = combine_frames(self.to_frame(), other.to_frame())
+        else:
+            combined = self._combine_frame_with_fill_value(other, fill_value=fill_value)
+
+        @pandas_udf(returnType=return_type)  # type: ignore
+        def wrapped_func(x: pd.Series, y: pd.Series) -> pd.Series:
+            return x.combine(y, func)
+
+        internal = InternalFrame(
+            spark_frame=combined._internal.spark_frame.select(
+                *combined._internal.index_spark_columns,
+                wrapped_func(*combined._internal.data_spark_columns),
+                NATURAL_ORDER_COLUMN_NAME
+            ),
+            index_spark_columns=combined._internal.index_spark_columns,
+            column_labels=self._internal.column_labels if np.isscalar(other) else [None],
+            column_label_names=[None],
+        )
+        return first_series(DataFrame(internal))
+
+    def _combine_frame_with_fill_value(self, other: "Series", fill_value: Any) -> DataFrame:
+        return combine_frames(
+            self._fill_value_for_missing_index(other, fill_value=fill_value),
+            other._fill_value_for_missing_index(self, fill_value=fill_value),
+        )
+
+    def _fill_value_for_missing_index(self, other: "Series", fill_value: Any) -> DataFrame:
+        filled_self = self._internal.spark_frame.select(
+            *self._internal.index_spark_column_names, *self._internal.data_spark_column_names
+        ).union(
+            other.index.difference(self.index)._internal.spark_frame.select(
+                *self._internal.index_spark_column_names, F.lit(fill_value)
+            )
+        )
+        return ps.DataFrame(
+            InternalFrame(
+                spark_frame=filled_self,
+                index_spark_columns=[
+                    scol_for(filled_self, col) for col in self._internal.index_spark_column_names
+                ],
+                index_names=self._internal.index_names,
+            )
+        )
```

Comment:
We can also leverage the `with_new_sdf` here.

```python
DataFrame(self._internal.with_new_sdf(filled_self))
```

Reply:
updated